### PR TITLE
Add TensorFlow 2.1.0

### PR DESCRIPTION
**Pull request by adamjstewart**
Successfully installs on macOS 10.15.2 with Clang 11.0.0 and Python 3.7.4.

If anyone else wants to test this for me, please do, as I don't have access to a GPU, aside from AWS, where I've never managed to get this to build.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-tensorflow/package.py [hunk 1]
```diff
@@ -58,9 +58,7 @@ class PyTensorflow(Package, CudaPackage):
 
     variant('mkl', default=False, description='Build with MKL support')
     variant('jemalloc', default=False, description='Build with jemalloc as malloc support')
-    # FIXME: ~gcp does not build for 2.0.0
-    # See https://github.com/tensorflow/tensorflow/issues/34878
-    variant('gcp', default=True, description='Build with Google Cloud Platform support')
+    variant('gcp', default=False, description='Build with Google Cloud Platform support')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorflow/package.py]: This works in 2.1.0!

File: var/spack/repos/builtin/packages/py-tensorflow/package.py [hunk 2]
```diff
@@ -669,10 +670,6 @@ def build(self, spec, prefix):
         if spec.satisfies('%gcc@5:'):
             args.append('--cxxopt=-D_GLIBCXX_USE_CXX11_ABI=0')
 
-        if spec.satisfies('@2.1:'):
-            # TODO: is this needed?
-            args.append('--define=tensorflow_mkldnn_contraction_kernel=0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorflow/package.py]: This wasn't needed to build 2.1.0.

**Comment by s-sajid-ali** [comment]
Isn't the required bazel version also bumped with each release ? 

I'm not sure where to look for this but the official build matrix does not list `2.1.0` as of now : `https://www.tensorflow.org/install/source#tested_build_configurations`

**Comment by s-sajid-ali** [comment]
Never mind, they've updated the build matrix for gpu builds and not the cpu builds. As of now the required bazel version remains the same (though cuDNN and cuda-toolkit versions are bumped).

**Comment by adamjstewart** [comment]
Yeah, we already had a lot of the changes figured out thanks to 2.1.0rc1.

**Pull request merged**